hyperspy/io_plugins/tvips.py

```diff
@@ -130,27 +196,152 @@ def file_reader(filename,
+        # save metadata
+        original_metadata = {'tvips_header': sarray2dict(header)}
+        # create custom dtype for memmap padding the frame_header as required
+        extra_bytes = increment - frame_header_dt.itemsize
+        record_dtype = TVIPS_RECORDER_FRAME_HEADER.copy()
+        record_dtype.append(("extra", bytes, extra_bytes))
+        record_dtype.append(("data", dtype, dimx*dimy))
 
+    # memmap the data
+    records_000 = np.memmap(filename, mode="r", dtype=record_dtype, offset=header["size"][0])
+    # the array data
+    all_array_data = [records_000["data"].reshape(-1, dimx, dimy)]
+    # in case we also want the frame header metadata later
+    metadata_keys = np.array(TVIPS_RECORDER_FRAME_HEADER)[:,0]
+    metadata_000 = records_000[metadata_keys]
+    all_metadata = [metadata_000]
+    # also load data from other files
+    for i in other_files:
+        # no offset on the other files
+        records = np.memmap(i, mode="r", dtype=record_dtype)
+        all_metadata.append(records[metadata_keys])
+        all_array_data.append(records["data"].reshape(-1, dimx, dimy))
+    if lazy:
+        data_stack = da.concatenate(all_array_data, axis=0)
+    else:
+        data_stack = np.concatenate(all_array_data, axis=0)
+    
+    # extracting some units/scales/offsets of the DP's or images
+    mode = all_metadata[0]["mode"][0]
+    DPU = "1/m" if mode == 2 else "m"
+    SDP = header["pixelsize"][0]
+    offsetx = header["offsetx"][0]
+    offsety = header["offsety"][0]
+    # modify the data if there is scan information
+    # we construct a 2D array of indices to slice the data_stack
+    if scan_shape is not None:
+        # try to deduce start and stop of the scan based on rotator index
+        if scan_shape == "auto":
+            record_idxs = np.concatenate([i["rotidx"] for i in all_metadata])
+            scan_start_frame, scan_stop_frame = _find_auto_scan_start_stop(record_idxs)
+            if scan_start_frame is None or scan_stop_frame is None:
+                raise ValueError("Scan start and stop information could not be automatically "
+                        "determined. Please supply a scan_shape and scan_start_frame.")
+            total_scan_frames = record_idxs[scan_stop_frame]  # last rotator
+            scan_dim = int(np.sqrt(total_scan_frames))
+            if not np.allclose(scan_dim, np.sqrt(total_scan_frames)):
+                raise ValueError("Scan was not square, please supply a scan_shape and start_frame.")
+            scan_shape = (scan_dim, scan_dim)
+            # there may be discontinuities which must be filled up
+            indices = _guess_scan_index_grid(record_idxs, scan_start_frame, scan_stop_frame)
+        # scan shape and start are provided
+        else:
+            total_scan_frames = scan_shape[0] * scan_shape[1]
+            indices = np.arange(scan_start_frame, scan_start_frame+total_scan_frames).reshape(scan_shape[0], scan_shape[1])
```

Comment:
```scan_shape``` could be adapted to take any number of dimensions, for example to load the signal as 1d. Something like this would seem to work for any dimensionality:
```
total_scan_frames = np.prod(scan_shape)
indices = np.arange(scan_start_frame, scan_start_frame+total_scan_frames).reshape(scan_shape)
```